test/OpenTelemetry.Tests/Implementation/Trace/ActivityExtensionsTest.cs

```diff
@@ -107,5 +107,24 @@ public void LastSetStatusWins()
 
             Assert.True(activity.GetStatus().IsOk);
         }
+
+        [Theory]
+        [InlineData(ActivityKind.Client, ActivityKind.Client)]
+        [InlineData(ActivityKind.Consumer, ActivityKind.Consumer)]
+        [InlineData(ActivityKind.Internal, ActivityKind.Internal)]
+        [InlineData(ActivityKind.Producer, ActivityKind.Producer)]
+        [InlineData(ActivityKind.Server, ActivityKind.Server)]
+        public void SetKind(ActivityKind input, ActivityKind expected)
+        {
+            using var openTelemetrySdk = OpenTelemetrySdk.EnableOpenTelemetry(b => b
+                                                   .AddActivitySource(ActivitySourceName));
+
+            using var source = new ActivitySource(ActivitySourceName);
+            using var activity = source.StartActivity(ActivityName);
```

Comment:
the issue of setting activity dont exist for activities created using activitySource. It only exists for activites created via new Activity(). Lets use that in unit test also to avoid any confusion.